shell/src/main/java/alluxio/shell/command/CheckConsistencyCommand.java

```diff
@@ -48,28 +59,65 @@ public String getCommandName() {
   public void run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI root = new AlluxioURI(args[0]);
+    checkConsistency(root, cl.hasOption("r"));
+  }
+
+  private void checkConsistency(AlluxioURI path, boolean repairConsistency) throws
+      AlluxioException, IOException {
     CheckConsistencyOptions options = CheckConsistencyOptions.defaults();
-    List<AlluxioURI> inconsistentUris = FileSystemUtils.checkConsistency(root, options);
+    List<AlluxioURI> inconsistentUris = FileSystemUtils.checkConsistency(path, options);
     if (inconsistentUris.isEmpty()) {
-      System.out.println(root + " is consistent with the under storage system.");
-    } else {
+      System.out.println(path + " is consistent with the under storage system.");
+      return;
+    }
+    if (!repairConsistency) {
       Collections.sort(inconsistentUris);
       System.out.println("The following files are inconsistent:");
       for (AlluxioURI uri : inconsistentUris) {
         System.out.println(uri);
       }
+    } else {
+      Collections.sort(inconsistentUris);
+      System.out.println(path + " has: " + inconsistentUris.size() + " inconsistent files.");
+      List<AlluxioURI> inconsistentDirs = new ArrayList<AlluxioURI>();
+      for (int i = 0; i < inconsistentUris.size(); i++) {
+        AlluxioURI inconsistentUri = inconsistentUris.get(i);
+        URIStatus status = mFileSystem.getStatus(inconsistentUri);
+        if (status.isFolder()) {
+          inconsistentDirs.add(inconsistentUri);
+          continue;
+        }
+        System.out.println("repairing path: " + inconsistentUri);
+        DeleteOptions deleteOptions = DeleteOptions.defaults().setAlluxioOnly(true);
+        mFileSystem.delete(inconsistentUri, deleteOptions);
+        if (mFileSystem.exists(inconsistentUri)) {
+          mFileSystem.listStatus(inconsistentUri, ListStatusOptions.defaults()
+              .setLoadMetadataType(LoadMetadataType.Always));
+        }
```

Comment:
This logic is not working the way you intend it to. `mFileSystem.exists` is the call that is fetching the metadata from the UFS (since the Alluxio copy is already deleted). `mFileSystem.listStatus` is unnecessary.